data_processing/maq_transform_deviceuse.R

```diff
@@ -13,52 +13,247 @@ deviceuse_key = c(
+                    str_detect(
+                        Device_Use.ol.duration,
+                        "Never|Less than 30 minutes|30 minutes to 1 hour"
+                    ),
+                na.rm =
+                    TRUE
+            ) > 0,
+            sf_maq_Q1_nomorethan1h_weekend_computer = sum(
+                str_detect(Device_Use.ol.description, "computer") &
+                    str_detect(Device_Use.ol.description, "weekend") &
+                    str_detect(Device_Use.ol.description, "Time spent") &
+                    str_detect(
+                        Device_Use.ol.duration,
+                        "Never|Less than 30 minutes|30 minutes to 1 hour"
+                    ),
+                na.rm =
+                    TRUE
+            ) > 0,
+            sf_maq_Q1_nomorethan1h_weekend_ebooks = sum(
+                str_detect(Device_Use.ol.description, "electronic books") &
+                    str_detect(Device_Use.ol.description, "weekend") &
+                    str_detect(Device_Use.ol.description, "Time spent") &
+                    str_detect(
+                        Device_Use.ol.duration,
+                        "Never|Less than 30 minutes|30 minutes to 1 hour"
+                    ),
+                na.rm =
+                    TRUE
+            ) > 0,
+            sf_maq_Q1_nomorethan1h_weekend_consolevideo = sum(
+                str_detect(Device_Use.ol.description, "console video") &
+                    str_detect(Device_Use.ol.description, "weekend") &
+                    str_detect(Device_Use.ol.description, "Time spent") &
+                    str_detect(
+                        Device_Use.ol.duration,
+                        "Never|Less than 30 minutes|30 minutes to 1 hour"
+                    ),
+                na.rm =
+                    TRUE
+            ) > 0,
+            sf_maq_Q1_nomorethan1h_weekend_mobiledevice = sum(
+                str_detect(Device_Use.ol.description, "mobile device") &
+                    str_detect(Device_Use.ol.description, "weekend") &
+                    str_detect(Device_Use.ol.description, "Time spent") &
+                    str_detect(
+                        Device_Use.ol.duration,
+                        "Never|Less than 30 minutes|30 minutes to 1 hour"
+                    ),
+                na.rm =
+                    TRUE
+            ) > 0,
+            sf_maq_Q1_nomorethan1h_weekend_smartphone = sum(
+                str_detect(Device_Use.ol.description, "smartphone") &
+                    str_detect(Device_Use.ol.description, "weekend") &
+                    str_detect(Device_Use.ol.description, "Time spent") &
+                    str_detect(
+                        Device_Use.ol.duration,
+                        "Never|Less than 30 minutes|30 minutes to 1 hour"
+                    ),
+                na.rm =
+                    TRUE
+            ) > 0,
+            sf_maq_Q1_nomorethan1h_weekend_virtualassistant = sum(
```

Comment:
Could you also add a variable that combines all these new ones?
`sf_maq_Q1_nomorethan1h_weekend_all = sf_maq_Q1_nomorethan1h_weekend_computer & sf_maq_Q1_nomorethan1h_weekend_ebooks & ...`

Reply:
yep!